class.eventcalendar.plugin.php

```diff
@@ -199,28 +200,33 @@ public function postController_beforeBodyInput_handler($sender) {
         $sender->addJsFile('eventcalendar.js', 'plugins/EventCalendar');
         $sender->addDefinition(
             'EventCalendarCategoryIDs',
-            json_encode(c('EventCalendar.CategoryIDs'))
+            json_encode(unserialize(c('EventCalendar.CategoryIDs')))
         );
 
-        $categoryID = $sender->Discussion->CategoryID;
-
+        $categoryID = $sender->Data['Category']['CategoryID'];
         // initially don't hide elements in allowed categories
         $cssClass = 'P EventCalendarInput';
-        if (!in_array($categoryID, c('EventCalendar.CategoryIDs'))) {
-            $cssClass .= ' Hidden';
+        if (!in_array($categoryID, unserialize(c('EventCalendar.CategoryIDs')))) {
+            $cssClass .= ' hidden';
         }
-
         $year = date('Y');
         $yearRange = $year.'-'.($year + 1);
-        $fields = explode(',', t('EventCalendar.DateOrder', 'month,day,year'));
+        $fields = explode(',', t('EventCalendar.DateOrder', 'day,month,year'));
 
         echo '<div class="', $cssClass, '">';
-        echo $sender->Form->label('Event Date', 'EventCalendarDate');
+        echo $sender->Form->label('Date de début', 'EventCalendarDate');
```

Comment:
Mixing languages inside of a plugin is a bad idea. People with an unsupported language would see that mixture. The Form->label() method is translating its text so if you translate "Event Date" in your language file, that translation will be used here

Reply:
I know, it's not clean, It was not my intention to publish this code.